src/aggregator/aggregation/gauge.go

```diff
@@ -52,8 +54,16 @@ func NewGauge(opts Options) Gauge {
 }
 
 // Update updates the gauge value.
-func (g *Gauge) Update(value float64) {
-	g.last = value
+func (g *Gauge) Update(timestamp time.Time, value float64) {
+	if g.lastAt.IsZero() || timestamp.After(g.lastAt) {
```

Comment:
There won't be any issues here from using `time.Time` will there? Just wondering since there were definitely some issues with using time.Time everywhere on the query side

Reply:
Nah should be fairly fast since we honestly don't have too many gauges overlapping each other thankfully.